Seedrs/salesforce
# (improvement) add method to publish to streaming api

## What does this PR do?

Add a new method (`SalesforceSync::API.publish_event`) to the API class to publish events to the Salesforce EventBus.

Update the default Salesforce API version to be used to **45.0**. According to the Salesforce API release notes, the changes from the version **36.0** to **45.0** doesn't affect this GEM and neither the platform sync. 

## How should this be manually tested?
- Set the platform GEMFILE to use this branch
- Run `bundle install` inside the platform container and rebuild the platform docker image
- Update the Salesforce credentials with a sandbox that has an event topic
- In the platform Rails console, publish an event to the Salesforce EventBus with: 
`SalesforceSync::API.publish_event(EVENT_NAME, PAYLOAD)`

To test the sync
- Set the environment variable `SALESFORCE_ACTIVE` to `true`
- Create a user with an email and see if it is synced to Salesforce

## Any background context you want to provide?

To publish to the EventBus using REST the minimum Salesforce API version it is **37.0**. 

## External context (JIRA, invision, etc)

[Jira EN-691](https://seedrs.atlassian.net/browse/EN-691)
